src/pages/concepts/message-bus.mdx

```diff
@@ -0,0 +1,97 @@
+import Wrapper from "../../timvir/wrapper";
+export default Wrapper;
+
+# Message Bus
+
+Components on each Timvir page have access to a page-wide message bus. This bus provides
+a standardized communication channel so that components can communicate with each
+other. The main purpose is to allow implementing micro-interactions that combine multiple
+components on a page.
+
+The bus uses [wonka](https://wonka.kitten.sh/) (a streams/observable library) internally. Components
+can subscribe to all messages that are sent to the bus, optionally filter them to receive
+only those they are interested in.
+
+## API
+
+### useContext
+
+The wonka [Subject](https://wonka.kitten.sh/api/sources#makesubject) is stored in a React Context.
+To access it use the `useContext()` function from `@timvir/core`.
+
+```typescript
+import { useContext } from "@timvir/core";
+
+function Component() {
+  const { bus } = useContext();
+}
+```
+
+Once you have access to the bus, you can subscribe to the messages. The following example
+stores the last message in local React state.
+
+```typescript
+import * as React from "react";
+import { useContext } from "@timvir/core";
+import { pipe, subscribe } from "wonka";
+
+function Component() {
+  const { bus } = useContext();
+
+  const [message, setMessage] = React.useState();
+  React.useEffect(() => pipe(bus.source, subscribe(setMessage)).unsubscribe, [bus]);
+}
+```
+
+### useProps / send
+
+One common use case is to (temporarily) override the props of another component on the page.
+For this purpose we have the `useProps()` hook and `send()` function.
+
+If a component has an `id` (the standard HTML id attribute, a string), then it can be
```

Comment:
I would recommend to not use `id`, as it can only appear once on a page. Use a `data-id` (or something) attribute instead. This would allow you to target multiple components with one trigger, which could come in useful if, for example, you would have a theme toggle and want to show all components on a page with that theme.

Which opens this up a little bit: if it's already a global bus, is it also using a global state? If I navigate to a different page and there are components with the same `data-id`, would they then also get that theme? That would be nice. Also if I navigate away and back.

Reply:
You can target multiple components by sending it to an interface which is implemented by all components you're trying to target. It's not really up to the sender to decide who gets to process messages, the components can decide that based on path/interface/member

But if you're trying to target a specific component, it has to have an id. The HTML id attribute has exactly that purpose, to uniquely identify elements on a page.

The bus is local to the page (it's provided by the Page component). It does not make sense to make it persistent across pages because you can't target components which do not exist yet.